### PR TITLE
Dockerfiles for Rails application (Apache and Nginx).

### PR DESCRIPTION
**Pull request by gtarique**
### Commands to build Docker images:
`cd k8ez/images/ruby`
`docker build -t railsapp-nginx-alpine -f nginx/Alpine/Dockerfile .`
`docker build -t railsapp-nginx-debian -f nginx/Debian/Dockerfile .`
`docker build -t railsapp-apache-debian -f apache/Debian/Dockerfile .`

### CODE REVIEW

File: images/ruby/Alpine/supervisord.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[supervisord]
+nodaemon=true
+logfile=/var/log/supervisor/supervisord.log ; (main log file;default $CWD/supervisord.log)
+
+[program:rails]
```
Comment (jacobbeasley) [inline images/ruby/Alpine/supervisord.conf]: Like php, we'd want to start 2 processes here: 
 - nginx - would serve static files and reverse proxy back to rails
 - rails (you ahve thi spart already)

Comment (gtarique) [reply images/ruby/Alpine/supervisord.conf]: @jacobbeasley made the changes. please review.

File: images/ruby/Alpine/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+# Copy the Nginx configuration file into the container
+COPY --chown=appuser:appuser ../Alpine/nginx.conf /etc/nginx/nginx.conf
+
+# Make start_rails.sh executable
+RUN chmod +x $INSTALL_PATH/start_rails.sh
+RUN chown -R appuser:appuser $INSTALL_PATH/start_rails.sh
+
+# Change to the non-root user
+USER appuser
```
Comment (jacobbeasley) [inline images/ruby/Alpine/Dockerfile]: id number

File: images/ruby/Alpine/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,47 @@
+user appuser;
+worker_processes auto;
+pid /run/nginx.pid;
+events {
+    worker_connections 1024;
+}
+error_log /var/log/nginx/error.log;
```
Comment (jacobbeasley) [inline images/ruby/Alpine/nginx.conf]: needs to be stdout

File: images/ruby/Alpine/nginx.conf [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+http {
+    include /etc/nginx/mime.types;
+    default_type application/octet-stream;
+
+    # Log files
+    access_log /var/log/nginx/access.log;
```
Comment (jacobbeasley) [inline images/ruby/Alpine/nginx.conf]: stdout

Comment (gtarique) [reply images/ruby/Alpine/nginx.conf]: Updated as per your feedback.

**Comment by sahilnarang1** [comment]
@jacobbeasley  I have tested. It is running with non root user. Also I have verified it is writing logs to stdout.

**Pull request merged**